### PR TITLE
ebpf_program_t.links should be initialized

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Resolves: #394

Signed-off-by: Alan Jowett <alanjo@microsoft.com>

### CODE REVIEW

File: libs/execution_context/ebpf_program.c [hunk 1]
```diff
@@ -249,6 +249,9 @@ ebpf_program_create(ebpf_program_t** program)
         goto Done;
     }
 
+    ebpf_list_initialize(&local_program->links);
+    ebpf_lock_create(&local_program->links_lock);
```
Comment (saxena-anurag) [inline libs/execution_context/ebpf_program.c:253]: This is already happening in `ebpf_program_initialize()`

Comment (Alan-Jowett) [reply libs/execution_context/ebpf_program.c:253]: ebpf_program_create then _ebpf_program_free and we call _ebpf_program_detach_links with a non-initialized value for program->links, triggering the verified failure.

Comment (dthaler) [reply libs/execution_context/ebpf_program.c:253]: Should lines 309-310 be removed then so we don't need to initialize these twice?

Comment (saxena-anurag) [reply libs/execution_context/ebpf_program.c:253]: I think that flow is only present in the test code (`ebpf_program_create()` -> `_ebpf_program_free()`). In the EC code, everywhere, if call to `ebpf_program_create()` succeeds, then `ebpf_program_initialize()` is called.

Agree we should remove this code from ebpf_program_initialize() as it is moved here.

**Review by dthaler — CHANGES_REQUESTED** [review]
Please address Anurag's comment by making the code match https://github.com/microsoft/ebpf-for-windows/pull/401/commits/59eeeab43a54ecd484ee2388c363e21c049151ca

**Review by saxena-anurag — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**